### PR TITLE
Auth docs

### PR DESCRIPTION
**Pull request by davidpmccormick**
Updating the `Auth` component docs in Cardigan. I'm not convinced this is the best place for this kind of documentation (not least because the component doesn't have a specific UI), but should we get it somewhere visible first, at least?

### CODE REVIEW

**Comment by jamesgorrie** [comment]
>  I'm not convinced this is the best place for this kind of documentation

The docs seem very very specific to the component (the states etc). I would say when working on this component, this is very useful.

What other contexts might you find the documentation useful, and we can think of how to put it there?

File: catalogue/webapp/components/Auth/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+   - A `codeVerifier` – which is a base64-encoded random string – and put this in `localStorage`.
+   - A `codeChallenge`, which is hashed, base64-encoded version of the `codeVerifier`. This becomes part of the `loginUrl` we create.
+   - Going to the `loginUrl`, we are sent to Cognito’s hosted UI and prompted to sign in with a username and password, then we’re redirected back to our app, with a `code` query parameter in the URL.
+   - A `useEffect` that’s listening for the code query parameter will then update the `authState` to `authorising`.
+2. When it’s then `authorising`:
+   - We send the `code` parameter from Cognito, along with our `codeVerifier` (from `localStorage`) in a `POST` request to the authorisation endpoint.
+   - If it’s successful, we are returned a `token` containing an `access_token`, an `id_token` and a `refresh_token` and we store this in `localStorage`.
+   - We then set the `authState` to `loggedIn`.
+3. When it’s `loggedIn` we don’t do anything
+<!-- 4. When it’s expired we use the `refresh_token` that was sent with the original `access_token` in order to get a new `access_token` before doing a call to any protected resources. -->
```
Comment (jamesgorrie) [inline catalogue/webapp/components/Auth/README.md]: Why commented?

File: catalogue/webapp/components/Auth/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+The `Auth` component exposes a render prop, that receives the `authState`, the `loginUrl`, and the `token` in an object argument, then the consuming component can determine what to display to the user based on the values of these properties.
```
Comment (jamesgorrie) [inline catalogue/webapp/components/Auth/README.md]: Maybe we could leave a link to this?
https://aws.amazon.com/blogs/mobile/understanding-amazon-cognito-user-pool-oauth-2-0-grants/

File: catalogue/webapp/components/Auth/README.md (hunk 1 above)
Comment (davidpmccormick) [reply catalogue/webapp/components/Auth/README.md]: I don't think we're _actually_ doing this yet?

File: catalogue/webapp/components/Auth/README.md (hunk 2 above)
Comment (davidpmccormick) [reply catalogue/webapp/components/Auth/README.md]: Good shout

**Pull request merged**